modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/metadata/ColocationGroup.java

```diff
@@ -77,6 +80,11 @@ public static ColocationGroup forSourceId(long sourceId) {
         return new ColocationGroup(new long[] {sourceId}, null, null);
     }
 
+    /** */
+    public static ColocationGroup create(long[] sourceIds, List<UUID> nodeIds, List<List<UUID>> assignments) {
+        return new ColocationGroup(Arrays.copyOf(sourceIds, sourceIds.length), nodeIds, assignments);
```

Comment:
`Arrays.copyOf(sourceIds, sourceIds.length)` -> `sourceIds.clone()`?
I don't like this method at all. Perhaps, colocation group constructor was private for a reason, now we have openned it. Also here we assume that sourceIds can be modified and should be copied, but leave nodeIds and assignments as is. 
I think it's better to create method in this class to filter assignments for node and return new instance of colocation group instead of creation colocation group in FragmentMapping (move logic of `local` method from `FragmentMapping` to this class).